engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerOverseer.java

```diff
@@ -58,17 +59,29 @@ public DockerOverseer(
         }));
         pos = stdout.length();
         this.client = client;
-        init.thenAccept(containerId ->
-                executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS)
-        );
+        init.thenAccept(containerId -> {
+            this.containerId = containerId;
+            executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS);
+        });
         this.init = init;
     }
 
     @Override
     public CompletableFuture<ExecutionOutcome> abort() {
-        return result.isDone() ? result : CompletableFuture.completedFuture(
-                new ExecutionOutcome(Stage.ABORT, "Abort is not yet supported for docker analysis", null)
-        );
+        return CompletableFuture.supplyAsync(() -> {
+            if (result.isDone()) {
+                return result.join();
+            } else {
+                try {
+                    client.stopContainerCmd(containerId).exec();
+                    client.removeContainerCmd(containerId).withRemoveVolumes(true).exec();
+                    return new ExecutionOutcome(Stage.ABORT, "Docker container aborted successfully", null);
+                } catch (Exception e) {
+                    log.error("Error stopping the Docker container: {}", e.getMessage());
+                    throw new RuntimeException("Error aborting Docker container: " + e.getMessage());
```

Comment:
Defect: Exception should not be thrown here.
Now, this is the case when outcome with stage="ABORT" / error="something" should be produced